chemprop/uncertainty/evaluator.py

```diff
@@ -1,93 +1,156 @@
-from abc import abstractmethod
+from abc import ABC, abstractmethod
 
 from torch import Tensor
 
 from chemprop.utils.registry import ClassRegistry
 
 
-class UncertaintyEvaluator:
-    """
-    A class for evaluating the effectiveness of uncertainty estimates with metrics.
-    """
+class EvaluatorBase(ABC):
+    """A :class:`EvaluatorBase` is a base class to evaluates the quality of uncertainty estimates using a specified metric."""
 
     @abstractmethod
-    def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Tensor:
+    def fit(self, *args, **kwargs) -> Tensor:
         """
         Evaluate the performance of uncertainty predictions against the model target values.
         """
+        pass
 
 
-UncertaintyEvaluatorRegistry = ClassRegistry[UncertaintyEvaluator]()
+UncertaintyEvaluatorRegistry = ClassRegistry[EvaluatorBase]()
 
 
-class MetricEvaluator(UncertaintyEvaluator):
-    """
-    A class for evaluating confidence estimates of classification and multiclass datasets using builtin evaluation metrics.
-    """
```

Comment:
@KnathanM can you add this back when you work on the implementation of `MetricEvaluator`?

> Any valid classification or multiclass metric. Because classification and multiclass outputs are inherently probabilistic, any metric used to assess them during training is appropriate to evaluate the confidences produced after calibration.